### PR TITLE
[Needs Review] Graph props Length, Distance and Path.

### PR DESCRIPTION
**Pull request by Julian-Mentasti**

### CODE REVIEW

File: graphrep.pl [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+% dis(I, N1, N2, N): the distance from N1 to N2 in graph I.
+% - I is the graphID
+% - N1/N2 is From/To Node name
+% - N is the distance
+dis(I, N1, N2, N) :-
+    graph(I, _, EDGES, _),
+    path(N1, N2, EDGES, [], N).
+
+% path(I, N1, N2, PATH): the path from N1 to N2 in graph I.
+% This is broken idk why I can find the length but not the path.....
```
Comment (Julian-Mentasti) [inline graphrep.pl]: Send help

Comment (reggiemcdonald) [reply graphrep.pl]:
```prolog
% Public Api
% Graph is the graph being searched
% From is the origin
% To is the destination
% Path is the path to the node, if one exists
findPathOuter(Graph, From, To, Path) :-
    graph(Graph, Nodes, _, _),
    findPath(From, To, Nodes, [], Path).

% ------- Private implmentation ------------
findNode(Name, [node(Name,Edges)|Rest], node(Name,Edges)).
findNode(Name, [node(Other,_)|Rest], N) :-
    not(Name == Other),
    findNode(Name, Rest, N).

findPath(From, From, _, _, [From]).
findPath(From, To, Nodes, Visited, [From|Path]) :-
    \+ member(From, Visited),
    findNode(From, Nodes, node(From, Edges)),
    findPathForNode(Edges, To, Nodes, [From|Visited], Path).

findPathForNode([dEdge(Start,Next)|Rest], To, Nodes, Visited, Path) :-
    findPath(Next, To, Nodes, Visited, Path).
findPathForNode([dEdge(_,_)|Rest], To, Nodes, Visited, Path) :-
    findPathForNode(Rest, To, Nodes, Visited, Path).
```

Comment (reggiemcdonald) [reply graphrep.pl]: Probably could name stuff better, but you get the drift

Comment (Julian-Mentasti) [reply graphrep.pl]: soooo many trampolines huh

File: graphrep.pl [hunk 2]
```diff
@@ -2,6 +2,8 @@
 % our approach uses a graph-term from.
 % Design initially borrowed from prologsite.
 
+% :- dynamc graph/3.
```
Comment (kydas) [inline graphrep.pl]: should remove this

**Review by kydas — CHANGES_REQUESTED** [review]
Looks good just the minor cleanup

File: graphrep.pl (hunk 2 above)
Comment (Julian-Mentasti) [reply graphrep.pl]: Fixed

**Pull request merged**